packages/app/src/systems/Chains/eth/utils/localStorage.ts

```diff
@@ -0,0 +1,35 @@
+const BLOCK_DATE_KEY_SUBSTRING = 'ethBlockDate-';
+const HASH_DONE_KEY_SUBSTRING = 'ethToFuelTx';
+
+export const TxCache = {
+  getBlockDate: (blockHash: string) => {
+    return localStorage.getItem(generateBlockDateKey(blockHash));
+  },
+  setBlockDate: (blockHash: string, blockDate: string) => {
+    return localStorage.setItem(generateBlockDateKey(blockHash), blockDate);
+  },
+  getTxIsDone: (blockHash: string) => {
+    return localStorage.getItem(generateHashDoneKey(blockHash));
+  },
+  setTxIsDone: (blockHash: string, isDone: string) => {
+    return localStorage.setItem(generateHashDoneKey(blockHash), isDone);
+  },
```

Comment:
why needs to pass `isDone` here if the function is made to set `isDone` ? you can just set `true` manually

```suggestion
  setTxIsDone: (blockHash: string) => {
    return localStorage.setItem(generateHashDoneKey(blockHash),'true');
  },
```